test/unit/storage/DataAccessorBasedStore.test.ts

```diff
@@ -84,48 +85,6 @@ class SimpleDataAccessor implements DataAccessor {
   }
 }
 
-class SimpleSuffixStrategy implements AuxiliaryStrategy {
-  private readonly suffix: string;
-
-  public constructor(suffix: string) {
-    this.suffix = suffix;
-  }
-
-  public usesOwnAuthorization(): boolean {
-    return true;
-  }
-
-  public getAuxiliaryIdentifier(identifier: ResourceIdentifier): ResourceIdentifier {
-    return { path: `${identifier.path}${this.suffix}` };
-  }
-
-  public getAuxiliaryIdentifiers(identifier: ResourceIdentifier): ResourceIdentifier[] {
-    return [ this.getAuxiliaryIdentifier(identifier) ];
-  }
-
-  public isAuxiliaryIdentifier(identifier: ResourceIdentifier): boolean {
-    return identifier.path.endsWith(this.suffix);
-  }
-
-  public getSubjectIdentifier(identifier: ResourceIdentifier): ResourceIdentifier {
-    return { path: identifier.path.slice(0, -this.suffix.length) };
-  }
-
-  public isRequiredInRoot(): boolean {
-    return false;
-  }
-
-  public async addMetadata(metadata: RepresentationMetadata): Promise<void> {
-    const identifier = { path: metadata.identifier.value };
-    // Random triple to test on
-    metadata.add(namedNode('AUXILIARY'), this.getAuxiliaryIdentifier(identifier).path);
-  }
-
-  public async validate(): Promise<void> {
-    // Always validates
-  }
-}
-
 describe('A DataAccessorBasedStore', (): void => {
```

Comment:
Comment is not related to this line. I'm actually surprised there are no changes being made to the `DataAccessorBasedStore` (which is why I put the comment here since there are no changes there I can attach this to 😄 ).

The idea was that when accessing a metadata resource, such as `http://localhost:3000/.meta`, this would return all metadata of the resource. But as it see it now I assume this will only return the data stored in that file in the case of a file backend, and error in the case of the memory/sparql backend as those won't have results for that URL. Doing a GET to that resource should return everything a `getMetadata` call would return on the subject resource. This also means PATCH is not going to work on those backends as that does a GET and a PUT.

Just fixing the GET is also not going to be enough, as again those 2 other backends handle metadata differently than the file backend. I did not test this but my guess is that the memory backend would create a new resource with its own metadata when writing to the metadata URL. This is where a `writeMetadata` function as mentioned in https://github.com/solid/community-server/issues/1027#issuecomment-961149384 would be needed. Although I would have it take a `RepresentationMetadata` object as input instead of a data stream to be consistent with `getMetadata`. It would then be the responsibility of the `DataAcessorBasedStore` to convert the incoming data stream to a metadata object if the target is a metadata resource.